library/core/src/any.rs

```diff
@@ -774,22 +803,17 @@ impl hash::Hash for TypeId {
         // - It is correct to do so -- only hashing a subset of `self` is still
         //   compatible with an `Eq` implementation that considers the entire
         //   value, as ours does.
-        self.t.1.hash(state);
+        let data =
+        // SAFETY: we know there are 8 bytes without provenance at this location
+            unsafe { crate::ptr::read_unaligned(self.data.as_ptr().cast::<u64>().offset(1)) };
+        data.hash(state);
```

Comment:
Should we really throw away half the hash?
IMO it's fine to just say: yes there is provenance here but we are libcore so we rely on private rustc knowledge that actually, transmuting pointers to integers is entirely fine.

Reply:
Oh, right, `t.1` is the second half... fair.